android/src/main/java/com/baseflow/permissionhandler/PermissionHandlerPlugin.java

```diff
@@ -565,6 +621,12 @@ private boolean openAppSettings() {
           permissionNames.add(Manifest.permission.WRITE_EXTERNAL_STORAGE);
         break;
 
+      case PERMISSION_GROUP_IGNORE_BATTERY_OPTIMIZATIONS:
+        if (hasPermissionInManifest(Manifest.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS))
+          if (VERSION.SDK_INT >= VERSION_CODES.M)
```

Comment:
Minor optimization here might be to switch the two if statements around. It seems to me that checking the permission manifest would be a heavier operation then checking the Android version. Maybe this would already suffice:

```
if (VERSION.SDK_INT >= VERSION_CODES.M && hasPermissionInManifest(Manifest.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS)) {
  permissionNames.add(Manifest.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS);
}
```

Reply:
Yes, that will be easier.